components/automate-cli/cmd/chef-automate/ha_node_add.go

```diff
@@ -45,260 +23,59 @@ func addNodeHACmd() *cobra.Command {
+	addNodeHACmd.PersistentFlags().IntVarP(&addDeleteNodeHACmdFlags.opensearchCount, "opensearch-count", "o", 0, "New opensearch instances to be added. Works with --aws-mode flag")
+	addNodeHACmd.PersistentFlags().IntVarP(&addDeleteNodeHACmdFlags.postgresqlCount, "postgresql-count", "p", 0, "New postgresql instances to be added. Works with --aws-mode flag")
+	addNodeHACmd.PersistentFlags().BoolVar(&addDeleteNodeHACmdFlags.onPremMode, "onprem-mode", false, "Use this flag if the deployment type is on prem")
+	addNodeHACmd.PersistentFlags().BoolVar(&addDeleteNodeHACmdFlags.awsMode, "aws-mode", false, "Use this flag if the deployment type is AWS")
 	addNodeHACmd.PersistentFlags().BoolVarP(&addDeleteNodeHACmdFlags.autoAccept, "auto-accept", "y", false, "auto-accept")
 
 	return addNodeHACmd
 }
 
 func runAddNodeHACmd(addDeleteNodeHACmdFlags *AddDeleteNodeHACmdFlags) func(c *cobra.Command, args []string) error {
 	return func(c *cobra.Command, args []string) error {
-		sshconfig := &SSHConfig{}
-		nodeAdder := NewAddNode(writer, *addDeleteNodeHACmdFlags, NewNodeUtils(), initConfigHabA2HAPathFlag.a2haDirPath, &fileutils.FileSystemUtils{}, NewSSHUtil(sshconfig))
-		return nodeAdder.Execute(c, args)
-	}
-}
-
-type AddNodeImpl struct {
-	config                  ExistingInfraConfigToml
-	copyConfigForUserPrompt ExistingInfraConfigToml
-	automateIpList          []string
-	chefServerIpList        []string
-	opensearchIpList        []string
-	postgresqlIp            []string
-	nodeUtils               NodeOpUtils
-	flags                   AddDeleteNodeHACmdFlags
-	configpath              string
-	terraformPath           string
-	writer                  *cli.Writer
-	fileutils               fileutils.FileUtils
-	sshUtil                 SSHUtil
-}
-
-func NewAddNode(writer *cli.Writer, flags AddDeleteNodeHACmdFlags, nodeUtils NodeOpUtils, haDirPath string, fileUtils fileutils.FileUtils, sshUtil SSHUtil) HAModifyAndDeploy {
-	return &AddNodeImpl{
-		flags:         flags,
-		writer:        writer,
-		nodeUtils:     nodeUtils,
-		configpath:    filepath.Join(haDirPath, "config.toml"),
-		terraformPath: filepath.Join(haDirPath, "terraform"),
-		fileutils:     fileUtils,
-		sshUtil:       sshUtil,
-	}
-}
-
-func (ani *AddNodeImpl) Execute(c *cobra.Command, args []string) error {
-	if !ani.nodeUtils.isA2HARBFileExist() {
-		return errors.New(AUTOMATE_HA_INVALID_BASTION)
-	}
-	if ani.flags.automateIp == "" &&
-		ani.flags.chefServerIp == "" &&
-		ani.flags.opensearchIp == "" &&
-		ani.flags.postgresqlIp == "" {
-		c.Help()
-		return status.New(status.InvalidCommandArgsError, "Please provide service name and ip address of the node which you want to add")
-	}
-	err := ani.validate()
-	if err != nil {
-		return err
-	}
-	err = ani.modifyConfig()
-	if err != nil {
-		return err
-	}
-	if !ani.flags.autoAccept {
-		res, err := ani.promptUserConfirmation()
+		configPath := filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "config.toml")
```

Comment:
Is this `config.toml` file available for older versions? What if the user tries to run the latest CLI over the old HA deployment?